src/cpp/src/visual_language/minicpm/classes.cpp

```diff
@@ -536,6 +537,52 @@ void adjust_pos_cache(
     }
 }
 
+/// @brief Check if universal tag is given.
+/// Check if native tag is given.
+/// Assert different tag aren't mixed.
+/// If no any tag, prepend universal image tag.
+/// If native tag, assume incremental image order.
+/// Else replace universal tags with native tags and save image order.
+std::pair<std::string, std::vector<size_t>> unify_prompt(const std::string& prompt, const std::string& native_tag, size_t n_new_images, size_t n_prev_images) {
```

Comment:
if instead of `const std::string& native_tag` we will pass `const std::regex& native_tag_pattern`, would it be a generic function applicable for other models as well?

Reply:
I had an idea to use a lambda function because phi3_v's tag depends on image id and `native_tag` is used not only for recovering `images_sequence` but also to insert such tags.

Adding Qwen2-VL will force me to do anything about it anyway so I didn't think about the problem in this PR.

Reply:
ok, lets keep it as an upcoming step.